htlcswitch/switch_test.go

```diff
@@ -1835,7 +1836,58 @@ func TestMultiHopPaymentForwardingEvents(t *testing.T) {
 		n.carolChannelLink,
 	)
 	firstHop := n.firstBobChannelLink.ShortChanID()
-	for i := 0; i < numPayments; i++ {
+	for i := 0; i < numPayments/2; i++ {
+		_, err := n.makePayment(
+			n.aliceServer, n.carolServer, firstHop, hops, finalAmt,
+			htlcAmt, totalTimelock,
+		).Wait(30 * time.Second)
+		if err != nil {
+			t.Fatalf("unable to send payment: %v", err)
+		}
+	}
+
+	bobLog, ok := n.bobServer.htlcSwitch.cfg.FwdingLog.(*mockForwardingLog)
+	if !ok {
+		t.Fatalf("mockForwardingLog assertion failed")
+	}
+
+	// After sending 5 of the payments, trigger the forwarding ticker, to
+	// make sure the events are properly flushed.
+	bobTicker := n.bobServer.htlcSwitch.cfg.FwdEventTicker.(*ticker.Mock)
+
+	// We'll trigger the ticker, and wait for the events to appear in Bob's
+	// forwarding log.
+	timeout := time.After(15 * time.Second)
+	for {
+
+		select {
+		case bobTicker.Force <- time.Now():
+		case <-time.After(1 * time.Second):
+			t.Fatalf("unable to force tick")
+		}
+
+		// If all 5 events is found in Bob's log, we can break out and
+		// continue the test.
+		bobLog.Lock()
+		if len(bobLog.events) == 5 {
+			bobLog.Unlock()
+			break
+		}
+		bobLog.Unlock()
+
+		// Otherwise wait a little bit before checking again.
+		select {
+		case <-time.After(50 * time.Millisecond):
+		case <-timeout:
+			bobLog.Lock()
+			t.Fatalf("expected 5 events in event log, instead "+
+				"found: %v", spew.Sdump(bobLog.events))
+			bobLog.Unlock()
```

Comment:
Perhaps this should be deferred? The `t.Fatal` call _shouldn't_ allow it to get to this point.

Reply:
True, good catch :)